### PR TITLE
Add sound stuff + deps

### PR DESCRIPTION
**Pull request by sixtyfourktec**

### CODE REVIEW

**Review by jkloetzke — COMMENTED** [review]
I'd like to put alsa-lib into `recipes/libs` because it's really just a library. All other sound stuff should go to the "multimedia" category. Especially pipewire because it's not just about sound... ;-)

File: recipes/devel/lua.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    Libs: -L\${libdir} -llua -lm $LUA_LIBS
+    Cflags: -I\${includedir}
+    EOF
+
+multiPackage:
+    dev:
+        packageScript: installPackageDev "$1/install/"
+        provideDeps: [ "*-dev" ]
+
+    tgt:
```
[the remark this answers is not in the document]
Comment (sixtyfourktec) [reply recipes/devel/lua.yaml]: Done. I removed the tgt package, because lua actually doesn't provide any shared libs.

**Comment by sixtyfourktec** [comment]
> I'd like to put alsa-lib into `recipes/libs` because it's really just a library. All other sound stuff should go to the "multimedia" category. Especially pipewire because it's not just about sound... ;-)

done

File: recipes/libs/alsa-lib.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+inherit: [autotools]
+
+metaEnvironment:
+    PKG_VERSION: "1.2.12"
+    PKG_LICENSE: "LGPL-2.1+ (library), GPL-2.0+ (aserver)"
```
Comment (jkloetzke) [inline recipes/libs/alsa-lib.yaml:5]: This smells like we should split the package, shouldn't we?

File: recipes/multimedia/pipewire.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    pipewire -1 pipewire -1 * - - audio,video PipeWire System Daemon
+    EOF
+
+multiPackage:
+    dev:
+        packageScript: mesonPackageDev
+        provideDeps: [ "*-dev" ]
+
+    tgt:
+        packageScript: mesonPackageTgt
```
Comment (jkloetzke) [inline recipes/multimedia/pipewire.yaml:88]: Would it make sense to split the package into the daemon with its modules and the library?

File: recipes/multimedia/pipewire.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        -Dsdl2=disabled \
+        -Dcompress-offload=disabled \
+        -Decho-cancel-webrtc=disabled \
+        -Draop=disabled \
+        -Dreadline=disabled \
+        -Dudev=enabled \
+        -Dpipewire-alsa=enabled -Dalsa=enabled
+
+    mkdir -p install/.bob
+    cat >install/.bob/pipewire.user-table <<EOF
```
Comment (jkloetzke) [inline recipes/multimedia/pipewire.yaml:78]: Again, we should have the infrastructure in place first.

File: recipes/libs/alsa-lib.yaml (hunk 2 above)
Comment (jkloetzke) [reply recipes/libs/alsa-lib.yaml:5]: Maybe we do it the other way around: explicitly only call `autotoolsPackageTgt` in the `packageScript`. Debian is not even packaging aserver. So I guess there is no real use of it anyway.

Comment (sixtyfourktec) [reply recipes/libs/alsa-lib.yaml]: I removed that license, because actually I didn't build aserver.

File: recipes/multimedia/pipewire.yaml (hunk 3 above)
Comment (sixtyfourktec) [reply recipes/multimedia/pipewire.yaml]: I did this and also for wireplumber.

**Comment by sixtyfourktec** [comment]
I did redo the virt::udev package, because I believe having the general deps in there was wrong. Please have another look.

**Comment by jkloetzke** [comment]
> I did redo the virt::udev package, because I believe having the general deps in there was wrong. Please have another look.

Yeah, that is certainly better. I added a fixup commit because I think this could be improved by removing duplicate stuff. There was also a `use: []` missing. Please have a look again and squash the fixup if you're ok with it.

**Comment by sixtyfourktec** [comment]
> > I did redo the virt::udev package, because I believe having the general deps in there was wrong. Please have another look.
> 
> Yeah, that is certainly better. I added a fixup commit because I think this could be improved by removing duplicate stuff. There was also a `use: []` missing. Please have a look again and squash the fixup if you're ok with it.

I squashed the working parts of your fixup.

**Comment by jkloetzke** [comment]
Thanks!

**Pull request merged**